### PR TITLE
cyclonedds-cxx: init at v0.10.4

### PR DESCRIPTION
**Pull request by Linbreux**
C++ bindings for cyclonedds

## Description of changes

Add new package `cyclonedds-cxx`, c++ bindings for cyclone.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by MikaelFangel** [comment]
Your current commit should be separated into one commit `maintainers: add <handle>` and one initializing the package. The commits should be in that particular order.

The placement of your package should be `pkgs/by-name/so/some-package/package.nix` if possible (:

**Comment by Linbreux** [comment]
@MikaelFangel your requested changes have been made. Let me know if something else still isn't right.

**Comment by Linbreux** [comment]
@MikaelFangel a friendly reminder :wink:

File: pkgs/by-name/cy/cyclonedds-cxx/package.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "cyclonedds-cxx";
+  version = "0.10.4";
+
+  src = fetchFromGitHub {
+    owner = "eclipse-cyclonedds";
+    repo = "cyclonedds-cxx";
+    rev = version;
+    sha256 = "sha256-/Bb4lhDeJFCZpsf+EfKSJpX5Xv5mFms5miw36be1goQ=";
```
Comment (MikaelFangel) [inline pkgs/by-name/cy/cyclonedds-cxx/package.nix:16]:
```suggestion
    hash = "sha256-/Bb4lhDeJFCZpsf+EfKSJpX5Xv5mFms5miw36be1goQ=";
```

File: pkgs/by-name/cy/cyclonedds-cxx/package.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ cyclonedds ];
+
+  meta = with lib;
+    {
+      description = "C++ binding for Eclipse Cyclone DDS";
+      homepage = "https://cyclonedds.io/";
+      license = with licenses; [ epl20 ];
```
Comment (MikaelFangel) [inline pkgs/by-name/cy/cyclonedds-cxx/package.nix:27]: As I see it you should also include the Apache2 license mentioned in their readme. (:

File: pkgs/by-name/cy/cyclonedds-cxx/package.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    rev = version;
+    sha256 = "sha256-/Bb4lhDeJFCZpsf+EfKSJpX5Xv5mFms5miw36be1goQ=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ cyclonedds ];
+
+  meta = with lib;
+    {
```
Comment (MikaelFangel) [inline pkgs/by-name/cy/cyclonedds-cxx/package.nix:24]:
```suggestion
  meta = with lib; {
```

**Review by MikaelFangel — APPROVED** [review]

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -10117,6 +10117,12 @@
     githubId = 36448130;
     name = "Michael Brantley";
   };
+  linbreux = {
+    email = "linbreux@gmail.com";
+    github = "linbreux";
```
Comment (pbsds) [inline maintainers/maintainer-list.nix:10122]:
```suggestion
    github = "Linbreux";
```

according to https://api.github.com/user/29354411

File: pkgs/by-name/cy/cyclonedds-cxx/package.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+, stdenv
+, fetchFromGitHub
+, cmake
+, cyclonedds
+}:
+
+stdenv.mkDerivation rec {
+  pname = "cyclonedds-cxx";
+  version = "0.10.4";
+
```
Comment (pbsds) [inline pkgs/by-name/cy/cyclonedds-cxx/package.nix:11]:
```suggestion

  outputs = [ "out" "dev" ];

```

**Pull request merged**